### PR TITLE
Fix the order of the function arguments for spread operation

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/parser/js/js-parser-expr.c [hunk 1]
```diff
@@ -1675,23 +1675,17 @@ parser_process_unary_expression (parser_context_t *context_p, /**< context */
             }
 
 #if ENABLED (JERRY_ES2015)
-            lexer_token_type_t type = context_p->token.type;
-            if (type == LEXER_THREE_DOTS)
+            if (context_p->token.type == LEXER_THREE_DOTS)
             {
-              spread_arguments++;
+              has_spread_element = true;
+              call_arguments++;
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser-expr.c]: Does this divide the maximum number of arguments by 2?

Comment (rerobika) [reply jerry-core/parser/js/js-parser-expr.c]: Currently (127 spreaded arguments) in the worst case it does. However the usage of spread operation in function argument list serves the purpose of reducing function arguments by packing them into iterable object. IMO it's fine for now.

Comment (zherczeg) [reply jerry-core/parser/js/js-parser-expr.c]: What about adding another byte argument to the call instructions which contains the number of spread opcodes?

Comment (rerobika) [reply jerry-core/parser/js/js-parser-expr.c]: IMO, currently only 1 byte number argument is available for CBC opcodes.

Comment (zherczeg) [reply jerry-core/parser/js/js-parser-expr.c]: (I can work on it later if you prefer that)

Comment (rerobika) [reply jerry-core/parser/js/js-parser-expr.c]: That could be useful later. Moreover, the maximum number of arguments is not defined in the standard.

**Pull request merged**